examples/classification/script.jl

```diff
@@ -0,0 +1,184 @@
+# ```math
+# f \sim \mathcal{GP}(0, k(\cdot, \cdot'))
+# ```
+# and sample a function `f`.
+
+k_true = [30.0, 0.5]
+kernel_true = k_true[1] * (SqExponentialKernel() ∘ ScaleTransform(k_true[2]))
+
+lgp = LatentGP(GP(kernel_true), BernoulliLikelihood(), 1e-12)
+x_true = 0:0.02:6
+f_true, y_true = rand(lgp(x_true))
+
+plot(x_true, f_true; seriescolor="red", label="")  # Plot the sampled function
+
+# Then, the output of this sampled function is pushed through a logistic sigmoid
+# `μ = σ(f)` to constrain the output to `[0, 1]`.
+
+μ = mean.(lgp.lik.(f_true))
+plot(x_true, μ; seriescolor="red", label="")
+
+# Finally, the outputs `y` of the process are sampled from a Bernoulli
+# distribution with mean `μ`. We're only interested in the outputs at a subset
+# of inputs `x`, so we first pick some random input locations and then find the
+# corresponding values for `y`.
+
+N = 30  # The number of training points
+mask = sample(1:length(x_true), N; replace=false, ordered=true)  # Subsample some input locations
+x, y = x_true[mask], y_true[mask]
+
+scatter(x, y; label="Sampled outputs")
+plot!(x_true, mean.(lgp.lik.(f_true)); seriescolor="red", label="True function")
+
+# ## Creating an SVGP
+#
+# Now that we have some data sampled from a generative model, we can try to recover the
+# true generative function with an SVGP classification model.
+#
+# For this, we shall use a mixture of
+# [ParameterHandling.jl](https://github.com/invenia/ParameterHandling.jl) to
+# deal with our constrained parameters and
+# [Optim.jl](https://julianlsolvers.github.io/Optim.jl/stable/) to perform
+# optimimisation.
+#
+# The required parameters for the SVGP are - the kernel hyperparameters `k`, the
+# inducing inputs `z` and the mean and covariance of the variational
+# distribution `q`; given by `m` and `A` respectively. ParameterHandling
+# provides an elegant way to deal with the constraints on these parameters,
+# since `k` must be positive and `A` must be positive definite. For more
+# details, see the
+# [ParameterHandling.jl](https://github.com/invenia/ParameterHandling.jl)
+# readme.
+
+# First, we need to define a quick and dirty positive definite matrix type for
+# ParameterHandling.jl - this code can safely be ignored.
+
+struct PDMatrix{TA}
+    A::TA
+end
+
+function pdmatrix(A::AbstractMatrix)
+    return PDMatrix(A)
+end
+
+function ParameterHandling.value(P::PDMatrix)
+    A = copy(P.A)
+    return A'A
+end
+
+function ParameterHandling.flatten(::Type{T}, P::PDMatrix) where {T}
+    v, unflatten_to_Array = flatten(T, P.A)
+    function unflatten_PDmatrix(v_new::Vector{T})
+        A = unflatten_to_Array(v_new)
+        return PDMatrix(A)
+    end
+    return v, unflatten_PDmatrix
+end
+#md nothing #hide
+
+# Initialise the parameters
+
+M = 15  # number of inducing points
+raw_initial_params = (
+    k=(var=positive(rand()), precision=positive(rand())),
+    z=bounded.(range(0.1; stop=5.9, length=M), 0.0, 6.0),  # constrain z to simplify optimisation
```

Comment:
how about:
```suggestion
    z=bounded.(range(0.1, 5.9; length=M), 0.0, 6.0),  # constrain z to simplify optimisation
```
feels a bit more natural to me